sympy/integrals/integrals.py

```diff
@@ -516,6 +517,24 @@ def try_meijerg(function, xab):
                             function = ret
                             continue
 
+            if not isinstance(antideriv,Integral):
+                sym = xab[0]
+                terms = tuple()
+                if antideriv.is_Add:
+                    terms = antideriv.args
+                else:
+                    terms = tuple((antideriv,))
+                for term in terms:
+                    argtan = []
+                    for part in term.atoms(tan) | term.atoms(cot):
+                        argtan.append(part.args)
+                    for arg in argtan:
+                        from sympy import solve
+                        a = arg[0]
+                        K = solve(a-pi/2,sym)[0]
+                        P = limit(term, sym, K, dir="-") - limit(term, sym, K, dir="+")
+                        antideriv += P*floor((a-pi/2)/pi)
```

Comment:
This will probably work only if `a` is a *linear* function of `x = xab[0]`. Then `a = b*x + c` and it is possible to find `K` without `solve` (which may be expensive in general).

Reply:
If it has more roots, then can't we add `floor`'s to remove discontinuity at all of those roots ?

Reply:
You could test expressions like `tan(2*atan(x**2 - x))` to see what would happen.